src/builder-utils.c

```diff
@@ -1615,6 +1615,95 @@ builder_maybe_host_spawnv (GFile                *dir,
   return flatpak_spawnv (dir, output, 0, error, argv);
 }
 
+int
+builder_get_all_checksums (const char *checksums[BUILDER_CHECKSUMS_LEN],
+                           GChecksumType checksums_type[BUILDER_CHECKSUMS_LEN],
+                           const char *md5,
+                           const char *sha1,
+                           const char *sha256,
+                           const char *sha512)
+{
+  int i = 0;
+
+  /* The order here is such that the first one is the
+     one to use by default if using a single one. That is
+     the longest checksum except it defaults to sha256
+     which was historically the one flatpak-builder used */
+
+  if (sha256 != NULL && *sha256 != 0)
+    {
+      checksums[i] = sha256;
+      checksums_type[i] = G_CHECKSUM_SHA256;
+      i++;
+    }
+
+  if (sha512 != NULL && *sha512 != 0)
+    {
+      checksums[i] = sha512;
+      checksums_type[i] = G_CHECKSUM_SHA512;
+      i++;
+    }
+
+  if (sha1 != NULL && *sha1 != 0)
+    {
+      checksums[i] = sha1;
+      checksums_type[i] = G_CHECKSUM_SHA1;
+      i++;
+    }
+
+  if (md5 != NULL && *md5 != 0)
+    {
+      checksums[i] = md5;
+      checksums_type[i] = G_CHECKSUM_MD5;
+      i++;
+    }
```

Comment:
Do you want to add SHA-384 support too? It got added in GLib 2.52.

Reply:
(Does anyone use it in the wild?)

Reply:
I initially added it, but removed it because it needs a newer glib than what we require (and is available in rhel7/xenial).

Reply:
Ah, that makes sense.